x-pack/docs/en/security/backup-security-configuration.asciidoc

```diff
@@ -0,0 +1,207 @@
+are identical on every node.
+
+Because changes to the configuration files usually require a node restart for
+the changes to take effect, this configuration method is reserved for settings
+that change rarely (or are required for cluster formation, or require file
+system privileges for modifications). In a typical deployment, once a desired
+configuration is attained, subsequent changes will be infrequent.
+
+Given this, it is recommended to backup after every modification. For the YAML
+text files you also have the option of storing diffs against a baseline, or
+checking them into `git`, therefore making the changes more easi to review by
+the {es} administrator. The binary and encoded files, such as the private keys
+or the secure store file are extra sensitive; they require restricted read
+access.
+
+[float]
+==== Backup of API Based Configuration
+
+The backup of API based configuration uses the <<modules-snapshots, regular
+snapshot functionality>>. The snapshot API saves the settings, the type
+mapping and the contents of any index, to a location of a cloud storage service
+or on a shared network file system. The snapshot operation can also optionally
+capture the persistent settings from the global cluster metadata.
+
+ . To use the snapshot API to backup `.security` first create a repository.
+Creating it requires the elevated `manage` cluster privilege, yet it's a
+relatively infrequent operation. The elevated privileges are required to prevent
+regular users from exfiltrating data to locations of their choosing.
+ . Afterwards, all snapshots can be created inside that repository by a user with
+the limited `snapshot_user` role. This role grants privileges to create
+snapshots in any existing repositories. It also grants privileges to list and
+read settings for all indices (but no data).
++
+--
+The following example creates a new user `snapshot_user` in the native realm:
+
+[source,js]
+--------------------------------------------------
+POST /_security/user/snapshot_user
+{
+  "password" : "secret",
+  "roles" : [ "snapshot_user" ]
+}
+--------------------------------------------------
+
+And the next one exemplifies the create snapshot operation, inside the
+`my_backup` repository:
+
+[source,js]
+--------------------------------------------------
+PUT /_snapshot/my_backup/snapshot_1
+{
+  "indices": ".security",
+  "include_global_state": true <1>
+}
+--------------------------------------------------
+
+<1> A few {security-features} settings are stored in the global cluster metadata as
+persistent settings. These are included in the snapshot by, but for facilitating
+change reviews as well as restore convenience, we can pull them out to stand in
+their own file.
+
+[source,shell]
+--------------------------------------------------
+curl -X GET -u elastic "localhost:9200/_cluster/settings?pretty&flat_settings" | jq '.persistent | with_entries(select(.key|startswith("xpack.security")))'
```

Comment:
We could maybe improve how `filter_path` works with `flat_settings`so that we don't expect users to rely on an external tool like `jq` to filter out specific settings by names.

Reply:
Great suggestion @tlrx, I have raised https://github.com/elastic/elasticsearch/issues/43290